### PR TITLE
First cut of 1.8 support

### PR DESCRIPTION
**Pull request by guineveresaenger**
Enabled node auth.
Removed all 1.5 versioned template folders.
Still need to remove references and some logic involving 1.5 in follow-up issue.
This PR will run up a 1.8 Kubernetes cluster.

### CODE REVIEW

File: hack/dockerdev [hunk 1]
```diff
@@ -65,7 +65,7 @@ done
 
 MYHOME=${MYHOME:-${HOME}}
 CONFIGFILE="${CONFIGFILE:-${MYHOME}/.kraken/config.yaml}"
-IMAGE="${IMAGE:-quay.io/samsung_cnct/k2-tools:latest}"
+IMAGE="${IMAGE:-quay.io/guineveresaenger/kraken-tools:latest}"
```
Comment (DStorck) [inline hack/dockerdev]: did you mean to leave this image ?

Comment (guineveresaenger) [reply hack/dockerdev]: Nope! Thank you for catching that!

File: ansible/roles/kraken.config/files/config.yaml [hunk 2]
```diff
@@ -165,7 +165,7 @@ definitions:
     - &defaultKube
       name: defaultKube
       kind: kubernetes
-      version: v1.7.8
+      version: v1.7.6
```
Comment (coffeepac) [inline ansible/roles/kraken.config/files/config.yaml]: default kubernetes version should be v1.8.x unless there is a compelling reason not to.

**Comment by joejulian** [comment]
I tested 1.8.1, 1.7.8, and 1.6.11 which each worked for me.

**Comment by guineveresaenger** [comment]
As of today, this code will bring up a healthy 1.8 cluster, and is backwards compatible with versions 1.6 and 1.7.
Followup: there are still some deprecated 1.5 references in kraken-lib that will need to be cleaned up.

**Comment by guineveresaenger** [comment]
Copypasta from Slack:

Apparently, the schema validation demands that the `kubeVersionedWeaveFabric` have *both* a `default` and a `versions` key.
This is fine if we have different WeaveFabrics. However, they are going to be all the same for supported k8s versions, which means I have to do something like this:
``` &kubeVersionedWeaveFabric
      name: kubeVersionedWeaveFabric
      kind: versionedFabric
      type: weave
      kubeVersion:
        default: *defaultWeaveFabric
        versions:
          v1.6: *defaultWeaveFabric
```
in order to satisfy the schema validation.

How should I proceed?

**Comment by davidewatson** [comment]
I may have discovered a problem when bringing up a cluster with an old config:

```
bash-4.3# k get pods --namespace=kube-networking
NAME          READY     STATUS             RESTARTS   AGE
canal-1dtl0   2/3       CrashLoopBackOff   8          18m
canal-34k5s   2/3       CrashLoopBackOff   8          18m
canal-4vwcz   2/3       CrashLoopBackOff   8          18m
canal-6jqtj   2/3       CrashLoopBackOff   8          18m
canal-7kp18   2/3       CrashLoopBackOff   8          18m
canal-jj66b   2/3       CrashLoopBackOff   8          18m
canal-k6nrj   2/3       CrashLoopBackOff   8          18m
canal-lh83l   2/3       CrashLoopBackOff   8          18m
canal-lks7r   2/3       CrashLoopBackOff   8          18m
canal-nnnqj   2/3       CrashLoopBackOff   8          18m
canal-p0r4h   2/3       CrashLoopBackOff   8          18m
canal-sfr9p   2/3       CrashLoopBackOff   8          18m
canal-t3qtt   2/3       CrashLoopBackOff   8          18m
canal-tkb8k   2/3       CrashLoopBackOff   8          18m
canal-vrqtr   2/3       CrashLoopBackOff   8          18m
bash-4.3# 
```

Will report back soon.

**Pull request merged**